libnavui-voice/src/main/java/com/mapbox/navigation/ui/voice/api/AudioFocusDelegate.kt

```diff
@@ -1,8 +1,18 @@
 package com.mapbox.navigation.ui.voice.api
 
-internal interface AudioFocusDelegate {
+/**
+ * An Api that allows you to interact with the audio focus
+ */
+interface AudioFocusDelegate {
 
-    fun requestFocus()
+    /**
+     * Request audio focus. Send a request to obtain the audio focus
+     * @return true on successful focus change request.
+     */
+    fun requestFocus(): Boolean
```

Comment:
Wondering if we should return an `int` with the different return values (as Android does) instead of a `Boolean` 🤔 

```java
* @return {@link #AUDIOFOCUS_REQUEST_FAILED}, {@link #AUDIOFOCUS_REQUEST_GRANTED}
*     or {@link #AUDIOFOCUS_REQUEST_DELAYED}.

Reply:
In current implementation we only handle result was focus granted/not granted and audio player plays or skips instruction. Not sure that we need such details for audio player. Also in my example:
system can grant focus, but it can be busy on phone call.